internal/asset/locator.go

```diff
@@ -0,0 +1,58 @@
+package asset
+
+import (
+	"github.com/google/go-containerregistry/pkg/name"
+
+	"github.com/buildpacks/pack/internal/paths"
+)
+
+// LocatorType represents the different locations an image may be retrieved from
+type LocatorType int
+
+const (
+	// InvalidLocator represents an unknown or un-parsable locator type
+	InvalidLocator LocatorType = iota
+	// URI locator paths represent a file://, http:// or https:// URI
+	URILocator
+	// FilepathLocator represents a local file path (this may be relative)
+	FilepathLocator
+	// ImageLocator represents a valid image that may be expanded to <org>/<repo>:<tag>
+	ImageLocator
+)
+
+// String represents each locator as a printable string
+func (l LocatorType) String() string {
+	return []string{
+		"InvalidLocator",
+		"URILocator",
+		"FilepathLocator",
+		"ImageLocator",
+	}[l]
+}
+
+// GetLocatorType parses a locator and returns the LocatorType
+func GetLocatorType(locator string, relativeBaseDir string) LocatorType {
```

Comment:
Is there some way we can combine this with the existing `locator` searching mechanism, or is it totally unfit/unintuitive to do so?

Reply:
Definitly could but it would likely take quite a bit of work here to pull out all the `buildpack` locator specific logic. Was thinking this would be part of the larger refactoring story.